### ISSUE TITLE
New case for apstra_datacenter_generic_system resource clear_cts_on_destroy attribute 

### ISSUE THREAD

**Issue by michaelbhughes** [Juniper/terraform-provider-apstra#782]
### Summary
I believe I found a new case for the `clear_cts_on_destroy` attribute to account for. I was running a plan/apply for the `apstra_datacenter_generic_system` resource and ran into the error below.

My understanding is that if `clear_cts_on_destroy` is true and Terraform is destroying a resource, then the resource should clear the CT assignments before attempting to destroy the generic system so we do not see errors.

Please take a look and let's chat about this.

### Example Outputs
```sh
...

apstra_datacenter_generic_system.external_generic_system["SE28"]: Destroying... [id=OuYtwXxrFEhI08HSlLU]
apstra_datacenter_generic_system.external_generic_system["SE29"]: Destroying... [id=W_oYJRNIwYdyEARMgX8]

...

│ Error: failed to delete generic system
│ 
│ failed to delete external system "W_oYJRNIwYdyEARMgX8" -
│ {"api_response":null,"config_blueprint_version":0,"errors":{"link_ids":"Deleting
│ all links forming a LAG is not allowed since the LAG has assigned
│ structures: ['connectivity template', 'VN endpoint']. Link ids:
│ ['PF-422\u003c-\u003eSE29(L2_Link)[1]',
│ 'PF-422\u003c-\u003eSE29(L2_Link)[2]']"},"error_code":422}

│ Error: failed to delete generic system
│ 
│ failed to delete external system "OuYtwXxrFEhI08HSlLU" -
│ {"api_response":null,"config_blueprint_version":0,"errors":{"link_ids":"Deleting
│ all links forming a LAG is not allowed since the LAG has assigned
│ structures: ['connectivity template', 'VN endpoint']. Link ids:
│ ['PF-421\u003c-\u003eSE28(L2_Link)[1]',
│ 'PF-421\u003c-\u003eSE28(L2_Link)[2]']"},"error_code":422}
```

### Task Details
![image](https://github.com/user-attachments/assets/79c64980-29d6-4aea-afd7-ff62f6c6c5b0)

```json
Detailed Status

{
  "error_code": 422,
  "errors": {
    "link_ids": "Deleting all links forming a LAG is not allowed since the LAG has assigned structures: ['connectivity template', 'VN endpoint']. Link ids: ['PF-422<->SE29(L2_Link)[1]', 'PF-422<->SE29(L2_Link)[2]']"
  }
}
```

```json
Request Data

{
  "url": "http://100.113.72.133/api/blueprints/30607be7-7bb4-4624-8148-ce7f22c40fb0/delete-switch-system-links?async=full",
  "method": "POST",
  "data": {
    "link_ids": [
      "PF-422<->SE29(L2_Link)[2]",
      "PF-422<->SE29(L2_Link)[1]"
    ]
  },
  "headers": {
    "X-Forwarded-Proto": "https",
    "Accept": "application/json",
    "Accept-Encoding": "gzip",
    "X-Forwarded-For": "100.113.72.132",
    "X-User": "admin",
    "X-User-Id": "15bf7fb1-7486-4b4f-a9a4-bd4824a2ce80",
    "X-Real-Ip": "100.113.72.132",
    "Content-Length": "90",
    "Host": "100.113.72.133",
    "Connection": "close",
    "Content-Type": "application/json",
    "User-Agent": "terraform-provider-apstra/0.67.0"
  },
  "args": {
    "async": "full"
  }
}
```

**Comment by chrismarget-j** [comment]
Hi @michaelbhughes, thank you for opening this issue.

It looks like the problem is [here](https://github.com/Juniper/apstra-go-sdk/blob/bda42a1f3e8db0f4765347e548f91b830faf20ee/apstra/two_stage_l3_clos_switch_system_links.go#L263C1-L271C1).

```go
	// unpack the error
	var e struct {
		LinkIds []string `json:"link_ids"`
	}
	if json.Unmarshal(ds.Errors, &e) != nil {
		return err // unmarshal fail - surface the original error
	}
```

It seems that the JSON `link_ids` component of the error might be a string, and might be an *array* of strings, depending on how many links produced the error.

The code above is only prepared to unpack an array of errors (the condition we ran into when this problem cropped up previously). It did not anticipate that `link_ids` might be a standalone string.

**Comment by chrismarget-j** [comment]
This should be resolved in v0.68.3.
